lib/factory_girl/attribute_assigner.rb

```diff
@@ -53,7 +53,16 @@ def build_hash
     end
 
     def get(attribute_name)
-      @evaluator.send(attribute_name)
+      case obj = @evaluator.send(attribute_name)
+      when FactoryGirl::Evaluator
+        obj.instance
```

Comment:
@joshuaclayton I am not sure how we could avoid `attr_reader :instance` and still have this work. The writer could be made private though.

Reply:
Right - `instance` (reader) wasn't public previously. I think with the change you've included to call `instance`, a public writer and a protected reader might be a good compromise. Thoughts?